src/EventService.Data/FileRepository.cs

```diff
@@ -5,79 +5,78 @@
+public class FileRepository : IFileRepository
 {
   private readonly IDataProvider _provider;
 
-  public EventFileRepository(
+  public FileRepository(
     IDataProvider provider)
   {
     _provider = provider;
   }
 
-  public async Task<List<Guid>> CreateAsync(List<DbEventFile> files)
-  {
-    if (files is null || !files.Any())
-    {
-      return null;
-    }
-
-    _provider.EventFiles.AddRange(files);
-    await _provider.SaveAsync();
-
+  public async Task<List<Guid>> CreateAsync(List<DbFile> files)
+  {
+    if (files is null || !files.Any())
+    {
+      return null;
+    }
+
+    _provider.Files.AddRange(files);
+    await _provider.SaveAsync();
+
     return files.ConvertAll(x => x.FileId);
-  }
-
+  }
+
   public async Task<bool> RemoveAsync(List<Guid> filesIds)
-  {
-    if (filesIds is null || !filesIds.Any())
-    {
-      return false;
-    }
-
-    _provider.EventFiles.RemoveRange(
-      await _provider.EventFiles
-        .Where(x => filesIds.Contains(x.FileId)).ToListAsync());
-
-    await _provider.SaveAsync();
-
+  {
+    if (filesIds is null || !filesIds.Any())
+    {
+      return false;
+    }
+
+    _provider.Files.RemoveRange(
+      await _provider.Files
+        .Where(x => filesIds.Contains(x.FileId)).ToListAsync());
+
+    await _provider.SaveAsync();
+
     return true;
-  }
-
-  public async Task<(List<DbEventFile>, int filesCount)> FindAsync(FindEventFilesFilter filter)
-  {
-    if (filter is null)
-    {
-      return default;
-    }
-
-    IQueryable<DbEventFile> dbFilesQuery = _provider.EventFiles
-      .AsNoTracking();
-
-    return (
-      await dbFilesQuery.Where(file => file.EventId == filter.EventId)
-        .Skip(filter.SkipCount).Take(filter.TakeCount).ToListAsync(),
-      await dbFilesQuery.CountAsync());
-  }
-
-  public Task<List<DbEventFile>> GetAsync(List<Guid> filesIds)
-  {
-    return _provider.EventFiles.AsNoTracking().Where(x => filesIds.Contains(x.FileId)).ToListAsync();
-  }
-
-  public async Task<bool> CheckEventFilesAsync(Guid eventId, List<Guid> filesIds)
-  {
-    if (filesIds is null || !filesIds.Any())
-    {
-      return false;
-    }
-
-    return !await _provider.EventFiles
-      .AnyAsync(p => filesIds.Contains(p.FileId) && p.EventId != eventId);
+  }
+
+  public async Task<(List<DbFile>, int filesCount)> FindAsync(FindFilesFilter filter)
+  {
+    if (filter is null)
+    {
+      return default;
+    }
+
+    IQueryable<DbFile> dbFilesQuery = _provider.Files.AsNoTracking();
+
+    return (
+      await dbFilesQuery.Where(file => file.EntityId == filter.EntityId)
+        .Skip(filter.SkipCount).Take(filter.TakeCount).ToListAsync(),
+      await dbFilesQuery.CountAsync());
+  }
+
+  public Task<List<DbFile>> GetAsync(List<Guid> filesIds)
+  {
+    return _provider.Files.AsNoTracking().Where(x => filesIds.Contains(x.FileId)).ToListAsync();
+  }
+
+  public async Task<bool> CheckFilesAsync(Guid entityId, List<Guid> filesIds)
```

Comment:
rename to describe actual checking. I can't say by the method name what is checking and what this method is supposed to return